bluemix/configuration/core_config/cf_config.go

```diff
@@ -153,6 +157,22 @@ func (c *cfConfig) writeRaw(cb func()) {
 	}
 }
 
+func (c *cfConfig) refreshToken(refreshToken string) (uaa.Token, error) {
+	auth := uaa.NewClient(uaa.DefaultConfig(c.data.AuthorizationEndpoint), rest.NewClient())
+	refreshedToken, err := auth.GetToken(uaa.RefreshTokenRequest(refreshToken))
+	if err != nil {
+		return uaa.Token{}, err
+	}
+
+	// return an error if refreshed token is invalid
+	refreshedTokenInfo := NewUAATokenInfo(refreshedToken.AccessToken)
+	if !refreshedTokenInfo.exists() {
```

Comment:
I'm not too familiar with CF tokens, so I'm not 100% what are the guaranteed required values present in a token. Will defer this to @boyang9527

Reply:
it is a JWT token with user information, expiration etc in it.